test/SparqlAntlrParserTest.cpp

```diff
@@ -803,80 +773,52 @@ TEST(SparqlParser, HavingCondition) {
 }
 
 TEST(SparqlParser, GroupGraphPattern) {
-  // The types of arguments to gmock matchers are automatically deduced. We
-  // explicitly specify the types here. This e.g. allows to write
-  // `GraphPattern(false, {}, subMatcher1, subMatcher2)` instead of
-  // `IsGraphPattern(false, vector<SparqlFilter>{}, std::tuple{subMatcher1,
-  // subMatcher2})`.
-  auto GraphPattern = [](bool optional, vector<SparqlFilter>&& filters,
-                         const auto&... childMatchers) {
-    return IsGraphPattern(optional, filters, std::tuple{childMatchers...});
-  };
-
-  auto expectGraphPattern = makeExpectCompleteParser<parseGroupGraphPattern>(
-      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}});
+  auto expectGraphPattern = ExpectCompleteParse<&Parser::groupGraphPattern>{
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
   auto expectGroupGraphPatternFails =
-      makeExpectParsingFails<parseGroupGraphPattern>();
+      makeExpectParsingFails<&Parser::groupGraphPattern>();
 
   // Test the Components alone.
-  expectGraphPattern("{ }", GraphPattern(false, {}));
+  expectGraphPattern("{ }", GraphPattern());
   expectGraphPattern(
       "{ { ?a ?b ?c } }",
-      GraphPattern(
-          false, {},
-          IsGroup(GraphPattern(false, {}, IsTriples({{"?a", "?b", "?c"}})))));
+      GraphPattern(GroupGraphPattern(IsTriples({{"?a", "?b", "?c"}}))));
   expectGraphPattern(
       "{ { ?a ?b ?c } UNION { ?d ?e ?f } }",
-      GraphPattern(
-          false, {},
-          IsUnion(GraphPattern(false, {}, IsTriples({{"?a", "?b", "?c"}})),
-                  GraphPattern(false, {}, IsTriples({{"?d", "?e", "?f"}})))));
+      GraphPattern(IsUnion(GraphPattern(IsTriples({{"?a", "?b", "?c"}})),
+                           GraphPattern(IsTriples({{"?d", "?e", "?f"}})))));
```

Comment:
We should take about names:
Aren't they all matchers, but some are called `NameOfThing` and some are called `IsNameOfThing`.
My spontaneous suggestion would be to call the matchers `Union` `Triples` `Minus`, pack them into a namespace `sparqlParserTestHelpers::matchers` and then 
(in this .cpp file)
use `namespace m = sparqlParserTestHlepers::matchers`

Then this becomes:
```
m::GraphPatterns(m::Union(m::GraphPattern(m::Triples(....))....
```

Reply:
And, what I also find:
There is a lot of overhead because of the fact, the there are `triples` and there are `triples that are the only child of a (group) graph pattern. Semantically (in the tests) we don't care about this distinction, 
but it currently requires a lot of syntax to make the compiler happy. as I understood it, this is directly the motivation for the `unwrapVariant` which we have to turn into a matcher, to make this more consistent.

Reply:
The inconsistency with matchers named `IsGraphPattern`/`GraphPattern` comes from the fact that I haven't really thought much about the names yet. The original matchers are `Is...`. Then matchers with the overloads through the struct are just missing the `Is`.